components/admin_console/admin_definition.jsx

```diff
@@ -430,6 +430,19 @@ export default {
                         placeholder: t('admin.service.siteURLExample'),
                         placeholder_default: 'E.g.: "http://example.com:8065"',
                     },
+                    {
+                        type: Constants.SettingsTypes.TYPE_BUTTON,
+                        key: 'TestSiteURL',
+                        action: testSiteURL,
```

Comment:
you can pass the site URL here (see `it.stateIsTrue` for example of how to access state)

Reply:
I'm not too sure how. `action` is sent to the `requestAction` property in the `RequestButton` element, and it gets always called with the 2 args I defined in `testSiteURL`.
As far as I understand, the `stateIsTrue` returns a `(config, state) -> ..`, which will end up [here](https://github.com/mattermost/mattermost-webapp/blob/bd15558ac2d9063d1874fe9f4ce55d559142c749/components/admin_console/schema_admin_settings.jsx#L334). I'm not sure how to follow the same pattern for the action, since it's sent directly to the `RequestButton`.
I considered checking the number of arguments in the `props.requestAction`, and if it's 2 keep it as it is, and 3 will include the state... but feels pretty fragile.
What do you think? I might be missing something here.

Reply:
Actually, I could add a 3rd argument to https://github.com/mattermost/mattermost-webapp/blob/bd15558ac2d9063d1874fe9f4ce55d559142c749/components/admin_console/request_button/request_button.jsx#L165 to include the `state`, so in the `action` I can use it.
Not sure how clean this is though, what do you think?

Reply:
@reflog The `this.state` in the `request_button.jsx` is not the same object as the one I had in the `schema_admin_settings.jsx`. The one in the schema admin settings is the state that stores the form data, so I could use it to get the 'dirty' value. In the request button file, the state is a local state for the button, so I don't have access to the data I need.
So adding a 3rd argument to the requestAction doesn't work. Right?
Any better idea?